win_etw_macros/src/lib.rs

```diff
@@ -395,20 +400,47 @@ fn trace_logging_events_core(attr: TokenStream, item_tokens: TokenStream) -> Tok
                 method.span(),
                 "Event metadata is too large to encode; reduce the complexity of this event.",
             ));
-            continue;
+            break;
         }
         let event_metadata_len_b0 = (event_metadata_len & 0xff) as u8;
         let event_metadata_len_b1 = (event_metadata_len >> 8) as u8;
         event_metadata[0] = parse_quote! { #event_metadata_len_b0 };
         event_metadata[1] = parse_quote! { #event_metadata_len_b1 };
 
-        let event_attrs = parse_event_attributes(&mut errors, &method.sig.ident, &method.attrs);
+        let event_attrs = parse_event_attributes(&method.sig.ident, &method.attrs);
+
+        if let Err(e) = event_attrs {
+            errors.push(e);
+            break;
+        }
+
+        let event_attrs = event_attrs.unwrap();
```

Comment:
This can be written without an `unwrap`, like so:

```
let event_attrs = match parse_event_attributes(...) {
  Ok(attrs) => attrs,
  Err(e) => {
    errors.push(e);
    break;
  }
};
```

This avoids the `unwrap` call. Even though we humans know that the unwrap will always succeed, it's good to use patterns that avoid the need to use it. (Changes to control flow have a way of breaking those assumptions, anyway.)

Reply:
Went with this as it lets us continue,

```rust
        let event_attrs = match event_attrs {
            Ok(attributes) => attributes,
            Err(e) => {
                errors.push(e);
                continue;
            }
        };
```